test/cql-pytest/cassandra_tests/validation/operations/select_single_column_relation_test.py

```diff
@@ -20,6 +20,33 @@ def testInvalidCollectionEqualityRelation(cql, test_keyspace):
         assert_invalid_message(cql, table, "Collection column 'd' (map<int, int>) cannot be restricted by a '=' relation",
                              "SELECT * FROM %s WHERE a = 0 AND d=?", {0: 0})
 
+def testInvalidCollectionIntegerOperandEqualityRelation(cql, test_keyspace):
+    with create_table(cql, test_keyspace, "(a int PRIMARY KEY, b set<int>, c list<int>, d map<int, int>)") as table:
+        execute(cql, table, "CREATE INDEX ON %s (b)")
+        execute(cql, table, "CREATE INDEX ON %s (c)")
+        execute(cql, table, "CREATE INDEX ON %s (d)")
```

Comment:
I don't understand why you needed to create secondary indexes on these columns. Indexing of collection columns is a big thing, and the index isn't even used for the restrictions you used... Maybe you were trying to prove that, indeed, index isn't relevant to these queries? If that's what you want to do (and I'm not sure why), it can be a second test. But the simplest most important test doesn't need the indexes.

Reply:
I wrote these tests using `testInvalidCollectionEqualityRelation` and `testInvalidCollectionNonEQRelation` as example.
Now checked - seems you're absolutely right and we do not need these indexes at all. Will fix, thanks.